services/minespace-web/src/HOC/AuthenticationGuard.js

```diff
@@ -24,77 +25,63 @@ import * as ENV from "@/constants/environment";
+      keycloak.login({
+        redirectUri: redirectUrl,
+        idpHint: KEYCLOAK.bceid_idpHint,
+      });
+    }
+
+    const authenticate = async () => {
+      const authenticationInProgressFlag = localStorage.getItem("authenticationInProgressFlag");
+      const token = keycloak.token ?? null;
       const { type } = queryString.parse(window.location.search);
 
-      if(keycloak.authenticated && !authenticatingFromCoreFlag && !type) {
-        localStorage.setItem("authenticatingFromCoreFlag", true);
-        await props.authenticateUser(keycloak.token)
-          .then(() => {
-            setAuthComplete(true);
-          })
-          .catch(() => {
-            localStorage.removeItem("authenticatingFromCoreFlag");
-          })
+      if (keycloak.authenticated && !authenticationInProgressFlag && !type) {
+        localStorage.setItem("authenticationInProgressFlag", true);
+        await props.authenticateUser(token).catch(() => {
+          localStorage.removeItem("authenticationInProgressFlag");
+        });
       }
-
       // standard Authentication flow on initial load,
       // if token exists, authenticate user.
       if (token && !props.isAuthenticated) {
-        await props
-          .getUserInfoFromToken(token)
-          .then(() => setAuthComplete(true));
-      } else {
-        setAuthComplete(true);
+        await props.getUserInfoFromToken(token);
       }
-    }
+    };
 
     useEffect(() => {
       authenticate();
     }, [keycloak.authenticated]);
 
-    useEffect(() => {
-      // get guid from pathname - props.location is not available at this level thus cannot directly access props.match.params.id
-      const guid = window.location.pathname
-        .split("/mines/")
-        .pop()
-        .split("/")[0];
-
-      const token = localStorage.getItem("jwt");
-      const { redirectingFromCore } = queryString.parse(window.location.search);
-      const redirectUrl = `${ENV.WINDOW_LOCATION}${route.MINE_DASHBOARD.dynamicRoute(guid)}`;
-
-      // all routing from core includes 'redirectingFromCore=true', if the user is not authenticated on MineSpace yet, redirect to the Keycloak Login
-      if (redirectingFromCore && !token) {
-        keycloak.login({
-          redirectUri: redirectUrl,
-          idpHint: KEYCLOAK.bceid_idpHint,
-        })
-      }
-    }, []);
-
-    const { redirectingFromCore } = queryString.parse(window.location.search);
-    const authenticatingFromCoreFlag = localStorage.getItem("authenticatingFromCoreFlag");
-    const fromCore = !redirectingFromCore && !authenticatingFromCoreFlag;
+    const authenticationInProgressFlag = localStorage.getItem("authenticationInProgressFlag");
+    const authenticated = props.isAuthenticated || isPublic || keycloak.authenticated;
+    const authInProgress = authenticationInProgressFlag || !initialized;
 
-    if (props.isAuthenticated || isPublic) {
+    if (authenticated) {
```

Comment:
oh boy, now I can't remember why I took out isPublic and if it was on purpose

Reply:
I'm not sure which (if any) pages are flagged with `isPublic`, but I think we might want to leave it in.

Reply:
Now I remember. 😅 
It's because `authenticated = props.isAuthenticated || isPublic || keycloak.authenticated;`
(I didn't take it out exactly, just moved it around a bit)

But I don't like how that doesn't fit the meaning of authenticated, so I'm going to rename the variable.